tests/test_install_aktualizr_and_update.sh

```diff
@@ -0,0 +1,42 @@
+#! /bin/bash
+
+set -exuo pipefail
+SCRIPT_DIR="$(dirname "$0")"
+TEST_INSTALL_DESTDIR=$(mktemp -d)/install
+./scripts/selfupdate_server.py 0 "$TEST_INSTALL_DESTDIR"&
+PORT=$("$SCRIPT_DIR/find_listening_port.sh" $!)
+trap 'kill %1' EXIT
+
+until curl 127.0.0.1:"$PORT" &> /dev/null; do
+  sleep 0.2
+done
+
+$1/src/aktualizr_primary/aktualizr --version
+
+$1/src/uptane_generator/uptane-generator generate --path "$TEST_INSTALL_DESTDIR/fake_root"
+$1/src/uptane_generator/uptane-generator image --path "$TEST_INSTALL_DESTDIR/fake_root" --targetname selfupdate_2.0 --filename "./tests/test_data/selfupdate_2.0" --hwid selfupdate-device
+$1/src/uptane_generator/uptane-generator addtarget --path "$TEST_INSTALL_DESTDIR/fake_root" --targetname selfupdate_2.0 --hwid selfupdate-device --serial 723f79763eda1c753ce565c16862c79acdde32eb922d6662f088083c51ffde66
+$1/src/uptane_generator/uptane-generator signtargets --path "$TEST_INSTALL_DESTDIR/fake_root"
+
+TEMP_DIR=$(mktemp -d)
+mkdir -m 700 -p "$TEMP_DIR/import"
+cp ./tests/test_data/prov_selfupdate/* "$TEMP_DIR/import"
+echo -e "[storage]\\npath = \"$TEMP_DIR\"\\n[import]\\nbase_path = \"$TEMP_DIR/import\"" > "$TEMP_DIR/conf.toml"
+echo -e "[tls]\\nserver = \"http://localhost:$PORT\"" >> "$TEMP_DIR/conf.toml"
+$1/src/aktualizr_primary/aktualizr -c ./tests/config/selfupdate.toml -c "$TEMP_DIR/conf.toml" once
+
+# check the updated file appeared in the installation directory and sha256sum matches expectation
+TEST_INSTALL_IMAGE_DIR=$TEMP_DIR/images
+if [ ! -d $TEST_INSTALL_IMAGE_DIR ]; then
+    echo "$TEST_INSTALL_IMAGE_DIR does not exist"
+    exit 1
+else
+    cd $TEST_INSTALL_IMAGE_DIR
+    for filename in `ls`; do
```

Comment:
We should know filename in advance based on the hash. I'm debating the best way to get the hash, though. It might be to run `aktualizr-info --director-target | jq <something>` or maybe there's a better way.

It'd also be better to avoid a `cd` in the script.